bpl/neutral_dixon_coles.py

```diff
@@ -0,0 +1,492 @@
+        self.std_defence = None
+        self.std_attack = None
+        self.mean_home_advantage = None
+        self.std_home_advantage = None
+        self._team_covariates_mean = None
+        self._team_covariates_std = None
+
+    # pylint: disable=too-many-locals
+    @staticmethod
+    def _model(
+        home_team: jnp.array,
+        away_team: jnp.array,
+        num_teams: int,
+        home_goals: Iterable[int],
+        away_goals: Iterable[int],
+        neutral_venue: Iterable[int],
+        team_covariates: Optional[np.array],
+    ):
+        std_home_advantage = numpyro.sample(
+            "std_home_advantage", dist.HalfNormal(scale=1.0)
+        )
+        std_attack = numpyro.sample("std_attack", dist.HalfNormal(scale=1.0))
+        std_defence = numpyro.sample("std_defence", dist.HalfNormal(scale=1.0))
+        mean_defence = numpyro.sample("mean_defence", dist.Normal(loc=0.0, scale=1.0))
+        corr_coef = numpyro.sample("corr_coef", dist.Normal(0.0, 1.0))
+
+        mean_home_advantage = numpyro.sample(
+            "mean_home_advantage", dist.Normal(0.1, 0.2)
+        )
+
+        u = numpyro.sample("u", dist.Beta(concentration1=2.0, concentration0=4.0))
+        rho = numpyro.deterministic("rho", 2.0 * u - 1.0)
+
+        if team_covariates is not None:
+            standardised_covariates = (
+                team_covariates - team_covariates.mean(axis=0)
+            ) / team_covariates.std(axis=0)
+            num_covariates = standardised_covariates.shape[1]
+
+            with numpyro.plate("covariates", num_covariates):
+                attack_coefficients = numpyro.sample(
+                    "attack_coefficients", dist.Normal(loc=0.0, scale=1.0)
+                )
+                defence_coefficients = numpyro.sample(
+                    "defence_coefficients", dist.Normal(loc=0.0, scale=1.0)
+                )
+
+            attack_prior_mean = jnp.matmul(
+                standardised_covariates, attack_coefficients[..., None]
+            ).squeeze(-1)
+            defence_prior_mean = mean_defence + jnp.matmul(
+                standardised_covariates, defence_coefficients[..., None]
+            ).squeeze(-1)
+        else:
+            attack_prior_mean = 0.0
+            defence_prior_mean = mean_defence
+
+        with numpyro.plate("teams", num_teams):
+            standardised_attack = numpyro.sample(
+                "standardised_attack", dist.Normal(loc=0.0, scale=1.0)
+            )
+            standardised_defence = numpyro.sample(
+                "standardised_defence",
+                dist.Normal(
+                    loc=rho * standardised_attack, scale=jnp.sqrt(1.0 - rho**2.0)
+                ),
+            )
+
+            with reparam(config={"home_attack_advantage": LocScaleReparam(centered=0)}):
+                home_attack_advantage = numpyro.sample(
+                    "home_attack_advantage",
+                    dist.Normal(mean_home_advantage, std_home_advantage),
```

Comment:
Shouldn't you rename these parameters to something like `mean_home_attack_adv` and `std_home_attack_adv`